test-suite-data/PURL-TYPES.maven.json

```diff
@@ -0,0 +1,121 @@
+    "description": "valid maven purl",
+    "purl": "pkg:maven/org.apache.commons/io@1.3.4",
+    "canonical_purl": "pkg:maven/org.apache.commons/io@1.3.4",
+    "type": "maven",
+    "namespace": "org.apache.commons",
+    "name": "io",
+    "version": "1.3.4",
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "basic valid maven purl without version",
+    "purl": "pkg:maven/org.apache.commons/io",
+    "canonical_purl": "pkg:maven/org.apache.commons/io",
+    "type": "maven",
+    "namespace": "org.apache.commons",
+    "name": "io",
+    "version": null,
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "maven often uses qualifiers",
+    "purl": "pkg:Maven/org.apache.xmlgraphics/batik-anim@1.9.1?classifier=sources&repositorY_url=repo.spring.io/release",
+    "canonical_purl": "pkg:maven/org.apache.xmlgraphics/batik-anim@1.9.1?classifier=sources&repository_url=repo.spring.io/release",
+    "type": "maven",
+    "namespace": "org.apache.xmlgraphics",
+    "name": "batik-anim",
+    "version": "1.9.1",
+    "qualifiers": {
+      "classifier": "sources",
+      "repository_url": "repo.spring.io/release"
+    },
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "maven pom reference",
+    "purl": "pkg:Maven/org.apache.xmlgraphics/batik-anim@1.9.1?extension=pom&repositorY_url=repo.spring.io/release",
+    "canonical_purl": "pkg:maven/org.apache.xmlgraphics/batik-anim@1.9.1?extension=pom&repository_url=repo.spring.io/release",
+    "type": "maven",
+    "namespace": "org.apache.xmlgraphics",
+    "name": "batik-anim",
+    "version": "1.9.1",
+    "qualifiers": {
+      "extension": "pom",
+      "repository_url": "repo.spring.io/release"
+    },
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "maven can come with a type qualifier",
+    "purl": "pkg:Maven/net.sf.jacob-project/jacob@1.14.3?classifier=x86&type=dll",
+    "canonical_purl": "pkg:maven/net.sf.jacob-project/jacob@1.14.3?classifier=x86&type=dll",
+    "type": "maven",
+    "namespace": "net.sf.jacob-project",
+    "name": "jacob",
+    "version": "1.14.3",
+    "qualifiers": {
+      "classifier": "x86",
+      "type": "dll"
+    },
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "a name is required",
+    "purl": "pkg:maven/@1.3.4",
+    "canonical_purl": "pkg:maven/@1.3.4",
+    "type": "maven",
+    "namespace": null,
+    "name": null,
+    "version": null,
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": true
+  },
+  {
+    "description": "double slash // after scheme is not significant",
+    "purl": "pkg://maven/org.apache.commons/io",
+    "canonical_purl": "pkg:maven/org.apache.commons/io",
+    "type": "maven",
+    "namespace": "org.apache.commons",
+    "name": "io",
+    "version": null,
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
```

Comment:
Some of these tests are duplicates of core tests and aren't related to the package type.